File: lecture_007(Strings)/l_003_checkpalindrome.cpp
#include<iostream>
#include<string>
using namespace std;

void display(string s) {
    for(int i=0; i<s.size(); i++) {
        cout<<s[i]+" ";
    }
    cout<<endl;
}
bool ispalindrome(string s) {
    int si = 0;
    int ei = s.size()-1;
    while(si<ei) {
        if(s[si] != s[ei]) {
            return false;
        }
        si++;
        ei--;
    }
    return true;
}
int main() {
    string s;
    cin>>s;
    cout<<ispalindrome(s)<<endl;

    return 0;
}

File: lecture_007(Strings)/l_003_checkpalindrome.java
import java.util.Scanner;
public class l_003_checkpalindrome {
    public static Scanner scn = new Scanner(System.in);
    public static void main(String[] args) {
        String s = scn.nextLine();
        System.out.println(isPalindrome(s));
    }
    public static void display(String s) {
        for(int i=0; i<s.length(); i++) {
            System.out.print(s.charAt(i));
        }
        System.out.println();
    }
    public static boolean isPalindrome(String s) {
        int si = 0;
        int ei = s.length()-1;
        while(si<ei) {
            if(s.charAt(si) != s.charAt(ei)) {
                return false;
            }
            si++;
            ei--;
        }
        return true;
    }
}